test/process_managers/process_manager_telemetry_test.exs

```diff
@@ -0,0 +1,316 @@
+                 process_manager_name: "ExamplePM",
+                 process_state: %ExamplePM{message: "init"},
+                 process_uuid: ^uuid,
+                 recorded_event: ^event
+               },
+               metadata
+             )
+
+      assert_receive {[:commanded, :process_manager, :handle, :stop], 2, _measurements, _metadata}
+
+      refute_receive {[:commanded, :process_manager, :handle, :exception], _, _measurements,
+                      _metadata}
+    end
+
+    test "emit `[:commanded, :process_manager, :handle, :stop]` event" do
+      uuid = UUID.uuid4()
+
+      {:ok, instance} = start_process_manager_instance(uuid)
+
+      event = to_recorded_event(%Event{message: "start", type: "ok"})
+
+      :ok = ProcessManagerInstance.process_event(instance, event)
+
+      assert_receive {[:commanded, :process_manager, :handle, :start], 1, _measurements,
+                      _metadata}
+
+      assert_receive {[:commanded, :process_manager, :handle, :stop], 2, measurements, metadata}
+
+      assert match?(%{duration: _}, measurements)
+      assert is_integer(measurements.duration)
+
+      assert match?(
+               %{
+                 application: App,
+                 process_manager_module: ExamplePM,
+                 process_manager_name: "ExamplePM",
+                 process_state: %ExamplePM{message: "init"},
+                 process_uuid: ^uuid,
+                 recorded_event: ^event,
+                 commands: [%Ok{message: "start"}]
+               },
+               metadata
+             )
+
+      refute_receive {[:commanded, :process_manager, :handle, :exception], _num, _measurements,
+                      _metadata}
+    end
+
+    test "emit `[:commanded, :process_manager, :handle, :stop]` with error event" do
+      uuid = UUID.uuid4()
+
+      {:ok, instance} = start_process_manager_instance(uuid)
+
+      event = to_recorded_event(%Event{message: "stop", type: "error"})
+
+      :ok = ProcessManagerInstance.process_event(instance, event)
+
+      assert_receive {[:commanded, :process_manager, :handle, :start], 1, _measurements,
+                      _metadata}
+
+      assert_receive {[:commanded, :process_manager, :handle, :stop], 2, measurements, metadata}
+
+      assert match?(%{duration: _}, measurements)
+      assert is_integer(measurements.duration)
+
+      assert match?(
+               %{
+                 application: App,
+                 process_manager_module: ExamplePM,
+                 process_manager_name: "ExamplePM",
+                 process_state: %ExamplePM{message: "init"},
+                 process_uuid: ^uuid,
+                 recorded_event: ^event,
+                 error: "stop"
+               },
+               metadata
+             )
+
+      refute_receive {[:commanded, :process_manager, :handle, :exception], _num, _measurements,
+                      _metadata}
+    end
+
+    test "events are emitted with discrete start/stop on retries" do
```

Comment:
this test doesn't cover all retry cases, covering every one would be a bit of a burden, but i'm happy to do it if we think its worthwhile.